sdk/storage/Azure.Storage.DataMovement/src/Shared/LocalTransferCheckpointer.cs

```diff
@@ -125,41 +126,63 @@ public LocalTransferCheckpointer(string folderPath)
             CancellationToken cancellationToken = default)
         {
             CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
-            if (_transferStates.TryGetValue(transferId, out var result))
+            if (_transferStates.TryGetValue(transferId, out JobPlanFile result))
             {
-                return Task.FromResult<int>(result.JobParts.Count);
+                return Task.FromResult(result.JobParts.Count);
             }
             throw Errors.MissingTransferIdCheckpointer(transferId);
         }
 
         /// <inheritdoc/>
-        public override async Task<Stream> ReadableStreamAsync(
+        public override async Task<Stream> ReadJobPlanFileAsync(
+            string transferId,
+            int offset,
+            int length,
+            CancellationToken cancellationToken = default)
+        {
+            Stream copiedStream = new MemoryStream(length);
```

Comment:
See if we can use our shared source `PooledMemoryStream` here so we're not naively allocating on what seems like a hot path.

Reply:
Switched this and reading job part plan files to use `PooledMemoryBuffer` which is probably an improvement. However, I would not consider reading checkpoint files a hot path really. We really only do it when listing transfers for resume or actually resuming a transfer (and only at the start).